src/main/java/seedu/address/logic/util/BalanceCalculationUtil.java

```diff
@@ -19,8 +20,36 @@
     /**
      * Returns an updated balance for {@code payer}
      */
-    public static Balance calculatePayerBalance(Amount amount, Person payer, UniquePersonList payees) {
-        int numberOfInvolvedPersons = numberOfInvolvedPersons(payees);
+    public static Balance calculatePayerBalance(String transactionType, Amount amount, Person payer,
+                                                int numberOfInvolvedPersons) {
+        switch (transactionType) {
+        case AddTransactionCommand.COMMAND_WORD:
+            return calculateAddTransactionPayerBalance(amount, payer, numberOfInvolvedPersons);
+        case DeleteTransactionCommand.COMMAND_WORD:
+            return calculateDeleteTransactionPayerBalance(amount, payer, numberOfInvolvedPersons);
```

Comment:
I don't understand the utility of adding the argument transactionType, instead of just calling the method calculateAddTransacCommand in the addTransaction method of the addressbook, and calling the method calculateDeleteTransaction.